test/unittest/models/unittest_models.cpp

```diff
@@ -40,11 +40,34 @@ IniWrapper reduce_mean_last("reduce_mean_last",
                               constant_loss,
                             });
 
+static std::unique_ptr<NeuralNetwork> makeMolAttention() {
+  std::unique_ptr<NeuralNetwork> nn(new NeuralNetwork());
+  nn->setProperty({"batch_size=3"});
+
+  auto outer_graph = makeGraph({
+    {"input", {"name=in3", "input_shape=1:1:5"}},
+    {"input", {"name=in2", "input_shape=1:4:6"}},
+    {"input", {"name=in1", "input_shape=1:1:6"}},
```

Comment:
Could we change this to fc to check if derivatives are correct?

Reply:
I will update the #1758 unit test to include fc layer for derivatives.